R/plots.R

```diff
@@ -274,17 +231,49 @@ use_regular_grid_plot <- function(x) {
 
 # ------------------------------------------------------------------------------
 
-plot_perf_vs_iter <- function(x, metric = NULL, width = NULL, eval_time = NULL) {
+process_autoplot_metrics <- function(x, metric, eval_time) {
+  met_set <- .get_tune_metrics(x)
+  any_dyn <- any(purrr::map_lgl(metric, ~ is_dyn(met_set, .x)))
+
+  x <- estimate_tune_results(x)
+
+  # This next line updates the .metric columns to be consistent with what the
+  # metric set produces. For example, in the data, there might be a value of
+  # "demographic_parity" but the metric set knows about
+  # "demographic_parity(cyl)". `paste_param_by()` makes sure that they both
+  # have the same value (if any).
+  x <- paste_param_by(x)
+
+  x <- x %>%
+    dplyr::filter(.metric %in% metric) %>%
+    dplyr::filter(!is.na(mean))
+
+  num_eval_times <- length(eval_time[!is.na(eval_time)])
+
+  if(any_dyn & num_eval_times > 0) {
+    x <- x %>%
+      dplyr::filter(.eval_time %in% eval_time) %>%
+      dplyr::mutate(
+        .metric =
+          dplyr::if_else(!is.na(.eval_time),
+                         paste0(.metric, " @", format(.eval_time, digits = 5)),
+                         .metric))
```

Comment:
```suggestion
      dplyr::mutate(
        .metric =
          dplyr::if_else(
            condition = !is.na(.eval_time),
            true = paste0(.metric, " @", format(.eval_time, digits = 5)),
            false = .metric
          )
      )
```

This change is only cosmetic, but I think it reads nicer now